frameworks/FSharp/giraffe/src/App/Program.fs

```diff
@@ -0,0 +1,72 @@
+module App.App
+
+open System
+open Microsoft.AspNetCore.Builder
+open Microsoft.AspNetCore.Hosting
+open Microsoft.Extensions.Logging
+open Microsoft.Extensions.DependencyInjection
+open Giraffe
+open Microsoft.AspNetCore.Http
+
+// ---------------------------------
+// Models
+// ---------------------------------
+[<CLIMutable>]
+type JsonMessage = { message : string }
+
+// ---------------------------------
+// Web app
+// ---------------------------------
+
+let jsonUtf8 (data:obj) : HttpHandler =
+    fun (next : HttpFunc) (ctx : HttpContext) ->
+        let bytes = Utf8Json.JsonSerializer.Serialize(data)
+        ctx.SetContentType "application/json"
+        ctx.WriteBytesAsync bytes
+
+let webApp =
+    choose [
+        GET >=>
+            choose [
+                route "/plaintext" >=> text "Hello, World!"
+                route "/json" >=> json { message = "Hello, World!" }
+                route "/jsonutf8" >=> jsonUtf8 { message = "Hello, World!" }
+            ]
+        setStatusCode 404 >=> text "Not Found" ]
+
+// ---------------------------------
+// Error handler
+// ---------------------------------
+
+let errorHandler (ex : Exception) (logger : ILogger) =
+    logger.LogError(EventId(), ex, "An unhandled exception has occurred while executing the request.")
+    clearResponse >=> setStatusCode 500 >=> text ex.Message
+
+// ---------------------------------
+// Config and Main
+// ---------------------------------
+
+let configureApp (app : IApplicationBuilder) =
+    let env = app.ApplicationServices.GetService<IHostingEnvironment>()
+    (match env.IsDevelopment() with
+    | true  -> app.UseDeveloperExceptionPage()
+    | false -> app.UseGiraffeErrorHandler errorHandler)      
+        .UseGiraffe(webApp)
+
+let configureServices (services : IServiceCollection) =    
+    services.AddGiraffe() |> ignore
+
+let configureLogging (builder : ILoggingBuilder) =
+    let filter (l : LogLevel) = l.Equals LogLevel.Error
+    builder.AddFilter(filter).AddConsole().AddDebug() |> ignore
+
+[<EntryPoint>]
+let main _ =    
+    WebHostBuilder()
+        .UseKestrel()        
+        .Configure(Action<IApplicationBuilder> configureApp)
+        .ConfigureServices(configureServices)
+        .ConfigureLogging(configureLogging)

```

Comment:
Don't configure logging
Rule 3.
> All test implementations must disable all disk logging. For many reasons, we expect all tests will run without writing logs to disk. Most importantly, the volume of requests is sufficiently high to fill up disks even with only a single line written to disk per request. Please disable all forms of disk logging. We recommend but do not require disabling console logging as well.

Every other framework disables logging completely so having it enabled will be a disadvatage